llvm/lib/IR/Verifier.cpp

```diff
@@ -5203,6 +5204,19 @@ void Verifier::visitCallsiteMetadata(Instruction &I, MDNode *MD) {
   visitCallStackMetadata(MD);
 }
 
+void Verifier::visitCalleeTypeMetadata(Instruction &I, MDNode *MD) {
+  Check(isa<CallBase>(I), "!callee_type metadata should only exist on calls",
+        &I);
+  for (const MDOperand &Op : MD->operands()) {
+    Check(isa<MDNode>(Op.get()),
+          "The callee_type metadata must be a list of type metadata nodes");
+    auto *TypeMD = cast<MDNode>(Op.get());
+    Check(hasGeneralizedMDString(TypeMD),
```

Comment:
This only checks the second operand. Do we need to verify other parts of the structure, e.g. that there are exactly two ops and that the first op is an integer? (Is the integer required to be zero as this is about function types?)